File: core/java/src/main/java/hara/lang/protocol/IToMutable.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(
    namespace = "std.protocol.itomutable", name = "IToMutable", parents = {"IPersistent"})
public interface IToMutable extends IPersistent {
  @HaraMethod(value = "to-mutable", arity = 1)
  IMutable toMutable();
}

File: core/rust/src/lang/protocol/itomutable.rs
use super::IPersistent;
use hara_protocol_macros::hara_protocol;

#[hara_protocol(
    namespace = "std.protocol.itomutable",
    name = "IToMutable",
    parents = ["IPersistent"]
)]
pub trait IToMutable: IPersistent {
    type Mutable;

    #[hara_method(value = "to-mutable", arity = 1)]
    fn to_mutable(&self) -> Self::Mutable;
}
